src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py

```diff
@@ -71,21 +67,26 @@ def _main(settings, optlist, args):
                             test_count = int(match.groups()[0])
                         break
 
-            result_url = test_executor.upload_file_to_storage(results_location, settings)
+            if settings.output_uri is not None:
+                result_url = test_executor.upload_file_to_storage(results_location, settings)
+            else:
+                result_url = None;
```

Comment:
You probably know python better than I, is this else clause actually doing anything?  (as I could access result_url before and it'd be None then too)

Reply:
Well, I am not an expert in Python but just tested and without the assignment in the else clause I got an `Unhandled error: local variable 'result_url' referenced before assignment` in the next lines.